### DISCUSSION TITLE
How to give the user a choice between 2 2FA - that is OTP and Webauthn?

### DISCUSSION THREAD

**Discussion by dblas** [keycloak/keycloak#19519] [Q&A]
Hello,
When you have 2 kinds of different 2FA configured, say OTP and a secure key, you'd like keycloak give you the choice between both and not systematically one of them. Because you can forget one and that's the reason why you configured 2 or more.

I tried different configurations - as password + 2FA - in the browser workflow but keycloak, version 21.0.2, always proposes the first 2FA in the sequence,  here, the OTP and **never** a dialog box.

Any hint?
Thank you,
db
<img width="975" alt="Howto propose choice between OTP and secure key" src="https://user-images.githubusercontent.com/6929433/229741453-8757b76b-5720-4e9a-add3-efcbb66becdc.png">

**Comment by darius-m** [comment]
Discussion #14988 was on a similar topic. You could have a flow that uses two alternative steps, one for each 2FA option, but the `user configured` step should likely not be present. I think it messes with how Alternative sub-flows or executions are evaluated (i.e., it basically already performs the user-configured step implicitly for the execution / first execution in the sub-flow).

**Comment by dblas** [comment]
Hello. Thank you for your answer. I tried your suggestion as well as the ones listed in the Discussion [#14988](https://github.com/keycloak/keycloak/discussions/14988).
Nix. Doesn't want to work at all.
I'll bounce off the discussion quoted above trying to find if we are talking about the same thing (a panel that displays both paths) and/or if 21.0.2 doesn't have some kind of a bug or missing something in a .ftl file.
It works well when asking between passwordless WebAuthn and password (both are alternative) but doesn't pop up anything when between OTP form and WebAuthn => always takes the OTP path.
Have a nice day.
db

**Reply by VonNao** [reply]
Did you ever find a solution to this by any chance? I got it working before Login but only with than i got the problem that neither webauth nor otp was correctly displayed in the names.
